pkg/controllers/orb/schedulinginputs.go

```diff
@@ -0,0 +1,485 @@
+
+// Construct and reduce the Scheduling Input down to what's minimally required for re-simulation
+func NewSchedulingInput(ctx context.Context, kubeClient client.Client, scheduledTime time.Time, pendingPods []*v1.Pod,
+	stateNodes []*state.StateNode, bindings map[types.NamespacedName]string, instanceTypes []*cloudprovider.InstanceType) SchedulingInput {
+	return SchedulingInput{
+		Timestamp:          scheduledTime,
+		PendingPods:        pendingPods,
+		StateNodesWithPods: newStateNodesWithPods(ctx, kubeClient, stateNodes),
+		Bindings:           bindings,
+		InstanceTypes:      instanceTypes,
+	}
+}
+
+func NewSchedulingInputReconstruct(timestamp time.Time, pendingPods []*v1.Pod, stateNodesWithPods []*StateNodeWithPods,
+	bindings map[types.NamespacedName]string, instanceTypes []*cloudprovider.InstanceType) *SchedulingInput {
+	return &SchedulingInput{
+		Timestamp:          timestamp,
+		PendingPods:        pendingPods,
+		StateNodesWithPods: stateNodesWithPods,
+		Bindings:           bindings,
+		InstanceTypes:      instanceTypes,
+	}
+}
+
+func (snp StateNodeWithPods) GetName() string {
+	if snp.Node == nil {
+		return snp.NodeClaim.GetName()
+	}
+	return snp.Node.GetName()
+}
+
+func (si *SchedulingInput) Reduce() {
+	si.PendingPods = reducePods(si.PendingPods)
+	si.InstanceTypes = reduceInstanceTypes(si.InstanceTypes)
+}
+
+func (si SchedulingInput) String() string {
+	return protoSchedulingInput(&si).String()
+}
+
+func newStateNodesWithPods(ctx context.Context, kubeClient client.Client, stateNodes []*state.StateNode) []*StateNodeWithPods {
+	stateNodesWithPods := []*StateNodeWithPods{}
+	for _, stateNode := range reduceStateNodes(stateNodes) {
+		pods, err := stateNode.Pods(ctx, kubeClient)
+		if err != nil {
+			pods = nil
+		}
+
+		stateNodesWithPods = append(stateNodesWithPods, &StateNodeWithPods{
+			Node:      stateNode.Node,
+			NodeClaim: stateNode.NodeClaim,
+			Pods:      reducePods(pods),
+		})
+	}
+	return stateNodesWithPods
+}
+
+/* Functions to reduce resources in Scheduling Inputs to the constituent parts we care to log / introspect */
+
+func reducePods(pods []*v1.Pod) []*v1.Pod {
+	reducedPods := []*v1.Pod{}
+	for _, pod := range pods {
+		reducedPod := &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      pod.Name,
+				Namespace: pod.Namespace,
+				UID:       pod.GetUID(),
+			},
+			Status: v1.PodStatus{
+				Phase:      pod.Status.Phase,
+				Conditions: reducePodConditions(pod.Status.Conditions),
+			},
+		}
+		reducedPods = append(reducedPods, reducedPod)
+	}
+	return reducedPods
+}
+
+func reducePodConditions(conditions []v1.PodCondition) []v1.PodCondition {
+	reducedConditions := []v1.PodCondition{}
+	for _, condition := range conditions {
+		reducedCondition := v1.PodCondition{
+			Type:    condition.Type,
+			Status:  condition.Status,
+			Reason:  condition.Reason,
+			Message: condition.Message,
+		}
+		reducedConditions = append(reducedConditions, reducedCondition)
+	}
+	return reducedConditions
+}
```

Comment:
There isn't any filtering here, could we just use `pod.Status.Conditions` on `schedulinginputs.go:145`?